Automattic/wp-calypso
# Migrate the RECEIVE_PLUGINS error notice and remove lib/sites-list

This PR removes the last bit of code that uses anything from `lib/sites-list`. Namely, showing a "compressed" error notice when fetching site plugins fails:

<img width="345" alt="screen shot 2018-04-16 at 22 49 14" src="https://user-images.githubusercontent.com/664258/38862378-0a39b1c2-4235-11e8-9678-ee13d4d91a0d.png">

The rest is a series of janitorial commits that remove the `lib/sites-list` code and most (not all) references from comments and docs.

The impact on bundle size is not bad:
```
chunk                            stat_size           parsed_size           gzip_size
async-load-design~plugins~stats     +313 B  (+0.5%)       +250 B  (+0.7%)      +65 B  (+1.0%)
devdocs                              -80 B  (-0.0%)        -80 B  (-0.0%)      -13 B  (-0.0%)
manifest                              +0 B                  +0 B                -3 B  (-0.1%)
build                              -9286 B  (-0.3%)      -4681 B  (-0.4%)    -1164 B  (-0.4%)
```

**How to test:**
1. Have a few Jetpack/Atomic sites.
2. Go to `/plugins` (all sites selected, shows a list of plugins from WP.org)
3. In the Network devtool, toggle offline mode so that REST requests will start fail
4. Click on "Manage"
5. Check that all requests to fetch site plugins failed and a compressed notice (Error fetching plugins on N sites) is shown.

Fixes #24144.
